test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+			}
+			return nil
+		}
+		return fmt.Errorf("timed out waiting for container cgroup values to match expected")
+	}
+	waitPodStatusResourcesEqualSpecResources := func() (*v1.Pod, error) {
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			pod, err := podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			differs := false
+			for idx, c := range pod.Spec.Containers {
+				if !cmp.Equal(c.Resources, *pod.Status.ContainerStatuses[idx].Resources) {
+					differs = true
+					break
+				}
+			}
+			if differs {
+				continue
+			}
+			return pod, nil
+		}
+		return nil, fmt.Errorf("timed out waiting for pod spec resources to match pod status resources")
+	}
+	rsErr := waitForContainerRestart()
+	framework.ExpectNoError(rsErr, "failed waiting for expected container restart")
+	// Wait for pod resource allocations to equal expected values after resize
+	resizedPod, aErr := waitPodAllocationsEqualsExpected()
+	framework.ExpectNoError(aErr, "failed to verify pod resource allocation values equals expected values")
+	// TODO(vinaykul,InPlacePodVerticalScaling): Remove this check once base-OS updates to containerd>=1.6.9
+	//                containerd needs to add CRI support before Beta (See Node KEP #2273)
+	if !isInPlaceResizeSupportedByRuntime(c, pod.Spec.NodeName) {
+		// Wait for PodSpec container resources to equal PodStatus container resources indicating resize is complete
+		rPod, rErr := waitPodStatusResourcesEqualSpecResources()
+		framework.ExpectNoError(rErr, "failed to verify pod spec resources equals pod status resources")
+
+		ginkgo.By("verifying pod status after resize")
+		verifyPodStatusResources(rPod, expectedContainers)
+	} else if !framework.NodeOSDistroIs("windows") {
+		// Wait for container cgroup values to equal expected cgroup values after resize
+		// only for containerd versions before 1.6.9
+		cErr := waitContainerCgroupValuesEqualsExpected()
+		framework.ExpectNoError(cErr, "failed to verify container cgroup values equals expected values")
+	}
+	return resizedPod
+}
+
+func doPodResizeTests() {
+	f := framework.NewDefaultFramework("pod-resize-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	var podClient *e2epod.PodClient
+	ginkgo.BeforeEach(func() {
+		podClient = e2epod.NewPodClient(f)
+	})
+
+	var cpuCap int64
+	type testCase struct {
+		name        string
+		containers  []TestContainerInfo
+		patchString string
+		expected    []TestContainerInfo
+	}
+
+	noRestart := v1.NotRequired
+	doRestart := v1.RestartContainer
+	tests := []testCase{
+		{
+			name: "Guaranteed QoS pod, one container - increase CPU & memory",
+			containers: []TestContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &ContainerResources{CPUReq: "100m", CPULim: "100m", MemReq: "200Mi", MemLim: "200Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &noRestart,
+				},
+			},
+			patchString: `{"spec":{"containers":[
```

Comment:
Can this be a `v1.Pod` instead, and serialize it to a patch in the test execution?

Reply:
Can we take this discussion on a refactoring PR using common library for e2e and e2e_node tests ? As it is now i found it easier to follow the tests, i am worried moving it on another patch during execution will make it harder to follow up, i believe.